test/functional/mining_basic.py

```diff
@@ -45,7 +46,7 @@ def set_test_params(self):
 
     def mine_chain(self):
         self.log.info('Create some old blocks')
-        for _ in range(200):
+        for t in range(TIME_GENESIS_BLOCK, TIME_GENESIS_BLOCK + 200 * 156, 156):
```

Comment:
937404046e8c0f16e2284783d7aac76b783a8eae: it doesn't do anything helpful on its own and doesn't help 20167 in any way either. why doing this?

Reply:
The blockversion test introduced in [bitcoin#20167](https://github.com/bitcoin/bitcoin/pull/20167) requires the definition of `t` (where `t` is time of last minted block) to feed into `extra_args` as `mocktime` ([source](https://github.com/bitcoin/bitcoin/pull/20167/files#diff-54aed8126fc19e55100de2f73c7ea9cac88ade36ead20734cf4298df043eff04R63-R68)).

The [amended logic](https://github.com/dashpay/dash/pull/4638/commits/525c12e60f502d6cdc4efd50a965ae8369868da1#diff-54aed8126fc19e55100de2f73c7ea9cac88ade36ead20734cf4298df043eff04R45-R47) as part of [dash#4638](https://github.com/dashpay/dash/pull/4638) does not define `t`. The change resolves that.

Reply:
not in our case, we simply bump by 156 and do not use `t` in any way as far as I can tell

Reply:
That's true, in the backport of [bitcoin#15383](https://github.com/dashpay/dash/pull/4638/commits/525c12e60f502d6cdc4efd50a965ae8369868da1), we use `bump_mocktime` instead of `setmocktime` (which required `t`) for the reasons you mentioned in your comment.

But the test _introduced_ in bitcoin#20167 requires `t` to set `mocktime` via launch arguments.

https://github.com/dashpay/dash/blob/8844b66f40831794987c579042b7e32a7c5ebbca/test/functional/mining_basic.py#L61

It's not there in the backport done prior because it wasn't needed then but it is needed now, so I added it back.

Reply:
Oh, I see now... Using `t` outside of the loop is super confusing imo.